packages/next/src/build/webpack/loaders/next-metadata-image-loader.ts

```diff
@@ -52,12 +54,57 @@ async function nextMetadataImageLoader(this: any, content: Buffer) {
   const pathnamePrefix = path.join(basePath, segment)
 
   if (isDynamicResource) {
+    const mod = await new Promise<webpack.NormalModule>((res, rej) => {
+      this.loadModule(
+        resourcePath,
+        (err: null | Error, _source: any, _sourceMap: any, module: any) => {
+          if (err) {
+            return rej(err)
+          }
+          res(module)
+        }
+      )
+    })
+
+    const exportedFieldsExcludingDefault =
+      mod.dependencies
+        ?.filter((dep) => {
+          return (
+            [
+              'HarmonyExportImportedSpecifierDependency',
+              'HarmonyExportSpecifierDependency',
+            ].includes(dep.constructor.name) &&
+            'name' in dep &&
+            dep.name !== 'default'
+          )
+        })
+        .map((dep: any) => {
+          return dep.name
+        }) || []
     // re-export and spread as `exportedImageData` to avoid non-exported error
     return `\
-    import * as exported from ${JSON.stringify(resourcePath)}
+    import {
+      ${exportedFieldsExcludingDefault
+        .map((field) => `${field} as _${field}`)
+        .join(',')}
+    } from ${JSON.stringify(
+      // This is an arbitrary resource query to ensure it's a new request, instead
+      // of sharing the same module with next-metadata-route-loader.
+      // Since here we only need export fields such as `size`, `alt` and
+      // `generateImageMetadata`, avoid sharing the same module can make this entry
+      // smaller.
+      resourcePath.split('?')[0] +
```

Comment:
do we need to preserve other queries along with metadata query if `resourcePath.split[1]` existed?

Reply:
Actually, I just noticed that the original import (`import * as exported from ${JSON.stringify(resourcePath)}`) will never contain a query: https://webpack.js.org/api/loaders/#thisresource